library/ssl_tls12_server.c

```diff
@@ -2880,12 +2880,12 @@ static int ssl_prepare_server_key_exchange(mbedtls_ssl_context *ssl,
          * after the call to ssl_prepare_server_key_exchange.
          * ssl_write_server_key_exchange also takes care of incrementing
          * ssl->out_msglen. */
-        if ((ret = mbedtls_pk_sign(mbedtls_ssl_own_key(ssl),
-                                   md_alg, hash, hashlen,
-                                   ssl->out_msg + ssl->out_msglen + 2,
-                                   out_buf_len - ssl->out_msglen - 2,
-                                   signature_len)) != 0) {
-            MBEDTLS_SSL_DEBUG_RET(1, "mbedtls_pk_sign", ret);
+        if ((ret = mbedtls_pk_sign_ext((mbedtls_pk_sigalg_t) sig_alg, mbedtls_ssl_own_key(ssl),
```

Comment:
Unnecessary cast. Conceptually, the job of `mbedtls_ssl_get_ciphersuite_sig_pk_alg` is to tell which algorithm to pass to sign/verify function, so it wouldn't make sense to require a cast except in the middle of an API transition. And we're tying to finish the API transition now.